src/wallet/readable.go

```diff
@@ -118,19 +153,32 @@ func LoadReadableWallet(filename string) (*ReadableWallet, error) {
 }
 
 // ToWallet convert readable wallet to Wallet
-func (rw *ReadableWallet) ToWallet() (Wallet, error) {
-	w, err := newWalletFromReadable(rw)
+func (rw *ReadableWallet) toWallet() (*Wallet, error) {
+	ets, err := rw.Entries.toWalletEntries(rw.isEncrypted())
 	if err != nil {
-		return Wallet{}, err
+		return nil, err
+	}
+
+	w := &Wallet{
+		Meta:    rw.Meta,
+		Entries: ets,
+	}
+
+	if err := w.validate(); err != nil {
+		return nil, fmt.Errorf("invalid wallet %s: %v", w.Filename(), err)
 	}
-	return *w, nil
+
+	return w, nil
 }
 
-// Save saves to filename
+// Save saves to filename, remove .bak file if exist
 func (rw *ReadableWallet) Save(filename string) error {
-	// logger.Info("Saving readable wallet to %s with filename %s", filename,
-	// 	self.Meta["filename"])
-	return file.SaveJSON(filename, rw, 0600)
+	if err := file.SaveJSON(filename, rw, 0600); err != nil {
```

Comment:
Make new method that does not save .bak file, instead of creating one then deleting it